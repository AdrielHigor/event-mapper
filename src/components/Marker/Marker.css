.marker {
  height: 25px;
  width: 25px;
  position: relative;
  top: -33px;
  left: -13px;
}