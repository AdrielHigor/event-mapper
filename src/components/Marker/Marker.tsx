import React from "react";
import "./Marker.css";

export interface IMarker {
  lat: number;
  lng: number;
  onClick: () => void;
}

const Marker = ({ lat, lng, onClick }: IMarker) => (
  <img
    key={`${lat}${lng}`}
    src={require("../../res/images/location-marker.png")}
    alt="Location Marker"
    className="marker"
    onClick={onClick}
  />
);

export default Marker;
