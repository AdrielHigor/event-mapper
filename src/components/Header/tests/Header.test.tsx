import renderer from "react-test-renderer";
import Marker, { IHeader } from "../Header";

jest.mock("../Header.css", () => ({}));

describe("Marker", () => {
  let component: renderer.ReactTestRenderer;

  const defaultProps: IHeader = {
    onHelpClick: jest.fn(),
  };

  beforeEach(() => {
    component = renderer.create(<Marker {...defaultProps} />);
  });

  it("should render correctly", () => {
    expect(component).toMatchSnapshot();
  });

  it("should call onHelpClick function when help button clicked", () => {
    const helpBtn = component.root.findByProps({
      className: "help-button",
    });
    helpBtn.props.onClick();

    expect(defaultProps.onHelpClick).toHaveBeenCalledTimes(1);
  });
});
