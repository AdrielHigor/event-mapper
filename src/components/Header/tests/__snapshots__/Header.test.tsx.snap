// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Marker should render correctly 1`] = `
<div
  className="header"
>
  <h2
    className="header-h2"
  >
    Event Mapper
  </h2>
  <div
    className="help-button"
    onClick={[MockFunction]}
  >
    <span>
      Ajuda
    </span>
  </div>
</div>
`;
